apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1,311 +1,487 @@
+                let service_name = (span
+                    .attributes
+                    .get(&SERVICE_NAME)
+                    .cloned()
+                    .unwrap_or_else(|| Value::String("unknown service".into()))
+                    .as_str())
+                .to_string();
+                vec![TreeData::QueryPlan(QueryPlanNode {
+                    node: Some(apollo_spaceport::trace::query_plan_node::Node::Fetch(
+                        Box::new(FetchNode {
+                            service_name,
+                            trace_parsing_failed,
+                            trace,
+                            sent_time_offset: span
+                                .attributes
+                                .get(&APOLLO_PRIVATE_SENT_TIME_OFFSET)
+                                .and_then(Self::extract_i64)
+                                .map(|f| f as u64)
+                                .unwrap_or_default(),
+                            sent_time: Some(span.start_time.into()),
+                            received_time: Some(span.end_time.into()),
+                        }),
+                    )),
+                })]
+            }
+            "flatten" => {
+                vec![TreeData::QueryPlan(QueryPlanNode {
+                    node: Some(apollo_spaceport::trace::query_plan_node::Node::Flatten(
+                        Box::new(FlattenNode {
+                            response_path: span
+                                .attributes
+                                .get(&APOLLO_PRIVATE_PATH)
+                                .and_then(Self::extract_string)
+                                .map(|v| {
+                                    v.split('/').filter(|v|!v.is_empty() && *v != "@").map(|v| {
+                                        if let Ok(index) = v.parse::<u32>() {
+                                            ResponsePathElement { id: Some(apollo_spaceport::trace::query_plan_node::response_path_element::Id::Index(index))}
+                                        } else {
+                                            ResponsePathElement { id: Some(apollo_spaceport::trace::query_plan_node::response_path_element::Id::FieldName(v.to_string())) }
+                                        }
+                                    }).collect()
+                                }).unwrap_or_default(),
+                            node: child_nodes
+                                .into_iter()
+                                .filter_map(|child| match child {
+                                    TreeData::QueryPlan(node) => Some(Box::new(node)),
+                                    _ => None,
+                                })
+                                .next(),
+                        }),
+                    )),
+                })]
+            }
+            "subgraph" => {
+                vec![TreeData::Trace(self.find_ftv1_trace(span))]
+            }
+            "supergraph" => {
```

Comment:
use the constant you deleted in the file where we create the span. It's better because if we change the name of the span at the creation we might break the apollo tracing.

Reply:
Fixed in [9c07453](https://github.com/apollographql/router/pull/1514/commits/9c074531a4a2cdf8b6a3c85683b175720f7933dd)
Also added some more constants.